service/history/ndc/history_replicator.go

```diff
@@ -100,6 +100,11 @@ type (
 			ctx context.Context,
 			request *historyservice.ReplicateEventsV2Request,
 		) error
+
+		ApplyHistoryEvents(
```

Comment:
can you use `ApplyEventBlobs` API?  or rename `ApplyEventBlobs` to `ApplyHistoryEvents`

Reply:
Sure. I renamed the ApplyEventBlobs and added additional validation.